src/controller/python/README.md

```diff
@@ -1,244 +1,255 @@
 
-Genrally, once build dependencies are satisfied you can build the `python`
-target.
+1. Install all necessary packages and prepare the build system. For more details, see the [BUILDING.md](/docs/BUILDING.md) documentation.
 
-Use `scripts/build_python.sh` or run something equivalent to:
+   ```
+   sudo apt-get update
+   sudo apt-get upgrade
 
-```sh
-gn gen out/python_lib
-ninja -C out/python_lib python
-```
+   sudo apt-get install git gcc g++ python pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev bluez
+   ```
 
-3. Install Chip Device Controller
+   If the Python CHIP controller is built on a Raspberry Pi, install additional packages and reboot the device:
 
-> Note: Python device controller is not versioned, so you need to uninstall the
-> old device controller before install the new one.
+   ```
+   sudo apt-get install pi-bluetooth
+   sudo reboot
+   ```
 
-> It is recommended to setup a separate clean virtual environment The
-> `scripts/build_python.sh` script sets up a python environment and installs the
-> WHL file.
+2. Clone the Project CHIP repository.
 
-```sh
-virtualenv out/python_env --clear
-source out/python_env/bin/activate
-pip install out/python_lib/controller/python/chip*.whl
-```
+   ```
+   git clone https://github.com/project-chip/connectedhomeip.git
+   ```
 
-The WHL file installation will:
+3. Enter the `connectedhomeip` directory.
 
--   Install the 'chip' module
--   create a `ENV/bin/chip-device-ctrl` script that provides an interactive
-    shell for the chip library
+   ```
+   cd connectedhomeip
+   ```
 
-## BLE Virtualization on Linux (Optional)
+4. Initialize the git submodules.
 
-If we would like to setup virtual BLE central and peripheral in the same
-machine, then Bluez setup
+   ```
+   git submodule update --init
+   ```
 
-```
-cd third_party/bluez/repo
-./bootstrap
-third_party/bluez/repo/configure --prefix=/usr --mandir=/usr/share/man --sysconfdir=/etc --localstatedir=/var --enable-experimental --with-systemdsystemunitdir=/lib/systemd/system --with-systemduserunitdir=/usr/lib/systemd --enable-deprecated --enable-testing --enable-tools
-make
-```
+5. Build and install the Python CHIP controller.
 
-Note: You also need to install several packages on RPi if you want to build
-bluez
+   ```
+   scripts/build_python.sh -m platform
+   ```
 
-```
-sudo apt-get install libtool m4 automake autotools-dev libudev-dev libical-dev libreadline-dev
-```
+   > Note: To get more details about available build configurations, run the following command:
+   > ```scripts/build_python.sh --help```
 
-Run bluetoothd:
+6. Activate the Python virtual environment.
 
-```
-sudo third_party/bluez/repo/src/bluetoothd --experimental --debug &
-```
+   ```
+   source out/python_env/bin/activate
+   ```
 
-Bring up two virtual ble interface:
+7. Run the Python CHIP controller.
```

Comment:
We're still in the "Checkout / Build / Install" section, so maybe we need a separate section `## Running the tool` before this step or before step 3?